osdep/terminal-unix.c

```diff
@@ -407,10 +409,16 @@ void terminal_setup_getch(struct input_ctx *ictx)
     if (mp_make_wakeup_pipe(death_pipe) < 0)
         return;
 
+    tty_in = tty_out = open("/dev/tty", O_RDWR);
+    if (tty_in < 0) {
+        tty_in = STDIN_FILENO;
+        tty_out = STDOUT_FILENO;
+    }
+
     // Disable reading from the terminal even if stdout is not a tty, to make
     //   mpv ... | less
     // do the right thing.
-    read_terminal = isatty(STDIN_FILENO) && isatty(STDOUT_FILENO);
+    read_terminal = isatty(tty_in) && isatty(tty_out);
```

Comment:
This breaks the use-case in the above comment. It might make sense to keep the ``isatty(STDOUT_FILENO)`` test, since both ``mpv ... | less`` and ``find ... | mpv --playlist=-`` will work as expected.

Reply:
Valid concern, I guess. Esp. since the use case is mentioned there in the comment. Otherwise if there are no additional issues found it's highly likely that it's getting to LGTM level.